pyomo/common/tee.py

```diff
@@ -667,15 +705,18 @@ def close(self, in_exception=False):
 
         # Join all stream processing threads
         _poll = _poll_interval
+        _timeout = 0.0
         FAIL = False
         while True:
             for th in self._threads:
                 th.join(_poll)
+                _timeout += _poll
             self._threads[:] = [th for th in self._threads if th.is_alive()]
             if not self._threads:
                 break
-            _poll *= 2
-            if _poll_timeout <= _poll < 2 * _poll_timeout:
+            if _poll < _poll_timeout:
+                _poll *= 2.0
+            if _poll_timeout * 0.5 <= _poll < _poll_timeout:
```

Comment:
Something I have wondered for a while - why the factors of 2? Is it arbitrarily decided?